CO-725: Meta XE API frontend application

magical opened the pull request:
Adapted from the OAuth2 consent demo app

magical committed:
Initial implementation

CO-725

magical committed:
Implement filtering and pagination

CO-725

magical committed:
Remove drop-down navigation menu

CO-725

magical committed:
Don't show other instances when filtering by instance

CO-725

magical committed:
Clean up

CO-725

Remove cookie and session configuration, since we aren't using cookies
or sessions. Remove SeaSurf, since we don't need CSRF protection.

magical committed:
Add README, fix Dockerfile

CO-725

jaredkosanovic commented on Dockerfile:
```diff
@@ -0,0 +1,21 @@
+FROM python:2.7-alpine
+MAINTAINER Andrew Ekstedt <ekstedta@oregonstate.edu>
```
Could you use [LABEL](https://docs.docker.com/engine/reference/builder/#label) here instead? https://docs.docker.com/engine/reference/builder/#maintainer-deprecated

magical replied:
Huh. When was maintainer deprecated?

jaredkosanovic replied:
@magical https://github.com/docker/docker/pull/25466

jaredkosanovic commented on README.md:
```diff
@@ -0,0 +1,69 @@
+Meta XE Frontend
```
Could you add a config.py example file to this repo?

jaredkosanovic commented on Dockerfile:
```diff
@@ -0,0 +1,21 @@
+FROM python:2.7-alpine
```
Could you add a [healthcheck](https://docs.docker.com/engine/reference/builder/#healthcheck) to this image?

jaredkosanovic commented on metaxe/templates/index.html:
```diff
@@ -0,0 +1,162 @@
+        </div>
+      </div>
+    </section>
+
```
Multiple blank lines

jaredkosanovic commented on metaxe/templates/index.html:
```diff
@@ -0,0 +1,162 @@
+            <div class="col-sm-1">
+              <div class="form-group">
+                <label><br>
+                <button class="btn btn-primary">Filter</button>
```
Maybe we should add a "clear" or "reset" button around this area to make the instance set to "Any" and clear the search box.

jaredkosanovic commented on requirements.txt:
```diff
@@ -0,0 +1,20 @@
+appdirs==1.4.3
```
As you suggested, I had to change this to appdirs==1.4.2 or else I couldn't build the docker image. It told me that 1.4.2 was already installed (maybe along with a different package?). I didn't see any issues with running 1.4.2.

floppyears replied:
I ran into this issue as well.

jaredkosanovic reviewed (changes requested):
Overall great work. I tested this by running it in a container and it looks great and functions well.

magical committed:
Remove appdirs version from requirements.txt

CO-725

We ran into a problem where an earlier version of appdirs was already
installed in the container base image so pip uninstalled it and
everything broke.

So don't specify a version.

magical committed:
Whitespace nitpicking

CO-725

magical committed:
Use LABEL instead of MAINTAINER

CO-725

MAINTAINER is apparently deprecated.

floppyears commented:
* When viewing the list of apps in any instance the distinction between the applications is not as easy on the eyes. Either adding some top & bottom margin or alternate color on the rows would help.

floppyears commented on metaxe/templates/index.html:
```diff
@@ -0,0 +1,162 @@
+
+  <body>
+    <header>
+      <a href="http://oregonstate.edu"><img id="logo" src="http://oregonstate.edu/images/logo.jpg?v=1" alt="Oregon State University" width="215" height="78"/></a>
```
Why not specify the width and height in css?

magical replied:
Those are from the OSU template I based this off of.

magical committed:
Add Docker health check

CO-725

magical committed:
Add reset button

CO-725

magical committed:
Stripe table rows

CO-725

It was a little difficult to tell which versions corresponded to which
apps when all the instances were displayed.

magical committed:
Add example config file

CO-725

magical committed:
Address some comments from the demo

- Rename the Filter button to Search

- Make the page title more descriptive

magical committed:
Add some expository text
